src/lib.rs

```diff
@@ -46,6 +46,7 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 // Disable warnings to not clutter code with cfg too much
 #![cfg_attr(not(all(feature = "alloc", feature = "lfn")), allow(dead_code, unused_imports))]
+#![deny(missing_debug_implementations)]
```

Comment:
I'm not sure if this is a good idea. 
> This lint is “allow” by default because adding Debug to all types can have a negative impact on compile time and code size. It also requires boilerplate to be added to every type, which can be an impediment.

Source: https://doc.rust-lang.org/stable/nightly-rustc/rustc_lint/builtin/static.MISSING_DEBUG_IMPLEMENTATIONS.html

Some structs that you added derive contains reference to the filesystem itself which means they can debug print a lot of unrelated garbage (e.g. when printing `File` `FileSystem` and underlying IO will be printed too). Perhaps some custom implementation that skips `FileSystem` would be better...